snowflake/src/test/java/org/apache/iceberg/snowflake/InMemoryFileIO.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.snowflake;
+
+import java.util.Map;
+import org.apache.iceberg.exceptions.NotFoundException;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.InMemoryInputFile;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+public class InMemoryFileIO implements FileIO {
```

Comment:
Isn't there already an in-memory FileIO? If not, can we add this in a more generic location so others can use it?

Reply:
Not as far as I can tell - looks like InMemoryInputFile and InMemoryOutputFile were introduced in https://github.com/apache/iceberg/commit/ed54952b51a6cdc7d2350b8223c276421845a8e2 without an accompanying FileIO implementation. I went ahead and moved this into `core/src/test/java/org/apache/iceberg/io/`, expanded the functionality to behave as one would expected with both `InMemoryInputFile` and `InMemoryOutputFile`, and added its own little unittest.